Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -2698,3 +2698,42 @@ Function Get-AVSVSANClusterUNMAPTRIM {
             Get-Cluster | Get-VsanClusterConfiguration | Select-Object Name, GuestTrimUnmap
         }
 }
+
+function Remove-CustomRole {
+    <#
+    .DESCRIPTION
+        This function allows customer to remove a custom role from the SDDC.
+        Useful in case of roles created with greater privileges than Cloudadmin that can no longer be removed from the UI.
+    #>
+
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    param (
+        [Parameter(Mandatory = $true,
+            HelpMessage = "The name of the role to remove, as displayed in the vCenter UI (case insensitive). This must be a custom role.")]
+        [string]
+        $roleInput
+    )
+    # Check if the role exists before attempting removal
+    $roleToRemove = Get-VIRole | Where-Object { $_.Description -eq $roleInput }
+
+    # Check if the role is in the $donotremovearray or is a System role
+    if ($roleToRemove.Count -eq 1) {
+        if ((Test-AVSProtectedObjectName -Name $roleToRemove.Name) -or $roleToRemove.IsSystem -eq $true) {
+            Write-Error "'$roleInput' is either System or Built-in. Removal not allowed."
+        }
+        else {
+            try {
+                Remove-VIRole -Role $roleToRemove -Confirm:$false -Force:$false
+                Write-Host "The role '$roleInput' has been removed."
+            }
+            catch {
+                Write-Error "Failed to remove the role '$roleInput'."  
+            }
+        }
+    }
+    else {
```

Comment:
It would be helpful to output lost of objects found that match so that customer can self-service or in case of escalation to us, help us remove the exact role they might need.

Reply:
done